### PR TITLE
misc sanitizer fixes etc

### PR DESCRIPTION
**Pull request by douzzer**

### CODE REVIEW

File: wolfcrypt/test/test.c [hunk 1]
```diff
@@ -11479,31 +11482,33 @@ WOLFSSL_TEST_SUBROUTINE int XChaCha20Poly1305_test(void) {
...
     if (XMEMCMP(buf2, Plaintext, sizeof Plaintext))
-        ERROR_OUT(-6844, out);
+        ERROR_OUT(-6845, out);
 
   out:
 
 #if defined(WOLFSSL_SMALL_STACK) && !defined(WOLFSSL_NO_MALLOC)
-    XFREE(buf1, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
-    XFREE(buf2, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+    if (buf1 != NULL)
```
Comment (dgarske) [inline wolfcrypt/test/test.c:11508]: FYI: The `XFREE` macro is supposed to always check for NULL.

File: wolfcrypt/test/test.c [hunk 2]
```diff
@@ -11479,31 +11482,33 @@ WOLFSSL_TEST_SUBROUTINE int XChaCha20Poly1305_test(void) {
                                        Key, sizeof Key);
 
     if (ret < 0)
-        ERROR_OUT(-6840, out);
-
-    if (XMEMCMP(buf1, Ciphertext, sizeof Plaintext))
         ERROR_OUT(-6841, out);
 
-    if (XMEMCMP(buf1 + sizeof Plaintext, Tag, CHACHA20_POLY1305_AEAD_AUTHTAG_SIZE))
+    if (XMEMCMP(buf1, Ciphertext, sizeof Ciphertext))
```
Comment (dgarske) [inline wolfcrypt/test/test.c:11487]: Please use `sizeof()`.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfcrypt/test/test.c (hunk 1 above)
Comment (douzzer) [reply wolfcrypt/test/test.c:11508]: yup am aware -- was just an abundance of caution, but it sortof noised up the code.  would be better without.

File: wolfcrypt/test/test.c (hunk 2 above)
Comment (douzzer) [reply wolfcrypt/test/test.c:11487]: that entire routine uses the `sizeof var` construct consistently, 21 instances of it.

**Review by dgarske — APPROVED** [review]

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**